Update disputes subsytems section. (#6329)
paritytech/polkadot: roadmap/implementers-guide/src/node/disputes/README.md

# Disputes Subsystems

> If approval voting finds an invalid candidate, a dispute is raised. The disputes
> subsystems make that possible:

Reviewer: ```suggestion
If approval voting finds an invalid candidate, a dispute is raised. The disputes
subsystems are concerned with the following:
```

1. Disputes can be raised
2. Disputes (votes) get propagated to all other validators
3. Votes get recorded as necessary
3. Nodes will participate in disputes in a sensible fashion

> 4. Finality is halted if a candidate on the chain is disputed

Reviewer 2: ```suggestion
4. Finality is stopped while a candidate is being disputed on chain
```

> 5. Chains can be reverted in case a dispute concluded invalid

Reviewer 2: ```suggestion
5. Chains can be reverted in case a dispute concludes invalid
```

> 6. Votes are provided to the provisioner so losing validators can be slashed on chain

Reviewer 2: I'd say something like
6. Votes are being imported on chain via the provisioner
7. Once a dispute concludes on chain, validators on the losing side can be slashed

Author: I compressed it a bit, as this is already outside the scope of the dispute subsystem's, in particular it should not be its own point.

Reviewer 2: Makes sense. I was wondering if slashing is the only reason to put them onchain. I think we want that also to reach consensus on what's concluded so we can revert on invalid disputes safely (and not revert only on some validators).

The dispute-coordinator subsystem interfaces with the provisioner and chain

> selection to make the bulk of this possible. dispute-distribution is concerned

Reviewer: ```suggestion
selection to make the bulk of this possible. `dispute-distribution` is concerned
```
